pkg/volume/vsphere_volume/vsphere_volume_util.go

```diff
@@ -73,15 +82,56 @@ func (util *VsphereDiskUtil) CreateVolume(v *vsphereVolumeProvisioner) (vmDiskPa
 	// the values to the cloud provider.
 	for parameter, value := range v.options.Parameters {
 		switch strings.ToLower(parameter) {
-		case "diskformat":
+		case diskformat:
 			volumeOptions.DiskFormat = value
 		case "datastore":
 			volumeOptions.Datastore = value
+		case Policy_HostFailuresToTolerate:
```

Comment:
Same opinion as Ritesh on this one.. The validation function just did another switch which you have already done here. Better to combine them.